### PR TITLE
Tools: Add Get-Migrations and update "migrations list" to show applied or pending

### LINKED ISSUE #577: Commands: A way to see current Migrations status
We should enable the following scenarios:
- What migration have been applied to the database?
- Which ones are pending?
- Do I have unscaffolded model changes?

### PR DESCRIPTION
**Pull request by bricelam**
Manually verified that this works as expected and is both forward (3.1 tools on 5.0 projects) and backward (5.0 tools on 3.1 projects) compatible.

Resolves #577

### CODE REVIEW

File: src/EFCore.Design/Design/Internal/MigrationsOperations.cs [hunk 1]
```diff
@@ -143,17 +143,48 @@ private string SubnamespaceFromOutputPath(string outputDir)
...
+            }
+
             var services = _servicesBuilder.Build(context);
             EnsureServices(services);
 
             var migrationsAssembly = services.GetRequiredService<IMigrationsAssembly>();
             var idGenerator = services.GetRequiredService<IMigrationsIdGenerator>();
+            var historyRepository = services.GetRequiredService<IHistoryRepository>();
+
+            List<string> appliedMigrations = null;
```
Comment (smitpatel) [inline src/EFCore.Design/Design/Internal/MigrationsOperations.cs:164]: May be a HashSet?

Comment (bricelam) [reply src/EFCore.Design/Design/Internal/MigrationsOperations.cs]: Updated. It should be using OrdianlIgnoreCase too

File: src/EFCore.Design/Properties/DesignStrings.resx [hunk 2]
```diff
@@ -1,17 +1,17 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <root>
-  <!--
-    Microsoft ResX Schema
-
+  <!-- 
+    Microsoft ResX Schema 
+
```
Comment (Saibamen) [inline src/EFCore.Design/Properties/DesignStrings.resx:5]: Delete trailing whitespaces

Comment (AndriySvyryd) [reply src/EFCore.Design/Properties/DesignStrings.resx:5]: VS adds them

Comment (bricelam) [reply src/EFCore.Design/Properties/DesignStrings.resx:5]: It's actually `ResXResourceWriter.Generate()` that puts them in

Comment (bricelam) [reply src/EFCore.Design/Properties/DesignStrings.resx:5]: And it's random comments like this that lead to random ideas like #22156 😁

Comment (Saibamen) [reply src/EFCore.Design/Properties/DesignStrings.resx:5]: @bricelam: Could you paste URL for this method? (or repository URL)
Maybe someone could fix this

Comment (bricelam) [reply src/EFCore.Design/Properties/DesignStrings.resx:5]: Looks like its fixed in .NET Core. lol, good luck getting it patched in .NET Framework.
https://github.com/dotnet/winforms/blob/master/src/System.Windows.Forms/src/System/Resources/ResXResourceWriter.cs

Comment (bricelam) [reply src/EFCore.Design/Properties/DesignStrings.resx:5]: And yes, it really is in System.Windows.Forms.dll 🤦‍♂️

**Pull request merged**